### PR TITLE
Module #3 React Group Project: Space Traveler's Hub App

### PR DESCRIPTION
**Pull request by karlavdelgadof**
## React Group Project: Space Traveler's Hub App :rocket: :astronaut: :gear: 

These are the proposed changes :recycle:  for this project :arrow_down: :

### Config & basic setup :hammer_and_wrench: :memo: 
- Created React App.
- Installed React Redux, Redux Logger and React Router.
- Created routes and view components (Rockets, Missions and My Profile):
   - Implemented <NavLink /> for the page navigation links and style active class to indicate which section/page user is currently on.
- Created directories for all Redux state slice files (rockets and missions).

### Redux :atom_symbol: 
- Implemented **Space X** :rocket: API fetch with [axios](https://axios-http.com/docs/intro) to get data from the Missions [endpoint ](https://api.spacexdata.com/v3/missions) :round_pushpin: and Rockets [endpoint ](https://api.spacexdata.com/v3/rockets) :round_pushpin: when a user navigates :globe_with_meridians: to the Missions/Rockets section.
- Wrote actions and reducers for booking rockets and joining missions
- Implemented useDispatch() to dispatch the actions (join/leave/cancel/reserve) with the correct rocket/mission ID as an argument.

### Render UI:lists :1234: 
- Used useSelector() Redux Hook to select the state slices and render lists of rockets and missions in corresponding routes. 
- Add styling following the provided UI wireframe.
- Used the React conditional rendering syntax to handle the buttons/badges change :recycle: in the missions/rockets items :heavy_check_mark: 

### Render UI: My Profile section :bust_in_silhouette: 
- Created `profile.js` component to display a 2 column layout that lists ONLY the rockets/joined missions by the user:
	-Implemented .filter() method to get the items that meet the condition (true/false)

- Added Tests for all app components :heavy_check_mark: 

### Additional Requirements :heavy_plus_sign: :memo: 

- Added a placeholder message for when the "My Missions" or "My Rockets" lists are empty :right_anger_bubble: 
- Added the "Cancel reservation" and "Leave Mission" buttons to the lists in the Profile view :100: 
- Added the "Learn more" button for each mission and rocket. Upon click, it opens a corresponding Wikipedia page in a new tab :star_struck: 
- Created buttons "Join a Mission" and "Reserve a Rocket" that takes the user to the corresponding view from Profile :heavy_check_mark:

### CODE REVIEW

File: src/__test__/render.test.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+  it('match snapshot for the Profile', () => {
+    const tree = TestRenderer.create(
+      <Provider store={store}>
+        <Profile />
+      </Provider>,
+    ).toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+});
```
Comment (gemmen29) [inline src/__test__/render.test.js]: - [x] Kindly add tests to test all features and components per this [requirement](https://github.com/microverseinc/curriculum-react-redux/blob/main/group-project/project_space_travelers_hub.md#:~:text=NOTE%3A%20Make%20sure%20to%20add%20test%20for%20all%20your%20components%20and%20functions.) not only test snapshots and make sure that all tests run successfully.

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/__test__/render.test.js (hunk 1 above)
Comment (karlavdelgadof) [reply src/__test__/render.test.js]: Done :heavy_check_mark:

File: src/__test__/render.test.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+  it('match snapshot for the Profile', () => {
+    const tree = render(
+      <BrowserRouter>
+        <Provider store={store}>
+          <Profile />
+        </Provider>
+      </BrowserRouter>,
+    );
+    expect(tree).toMatchSnapshot();
+  });
```
Comment (omar-muhamad) [inline src/__test__/render.test.js:102]: - [x] As the previous reviewer mentioned, Please add unit tests for all functions and features used in your components and redux slices too not only screenshot tests as you did. 

image

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @karlavdelgadof and @Alzubair98,

Nice job, you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- Gitflow is used correctly. ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/__test__/render.test.js (hunk 2 above)
Comment (Alzubair98) [reply src/__test__/render.test.js:102]: hi 
its done

**Review by ShoiraTa — APPROVED** [review]
## **STATUS: APPROVED  🟢 ** 
------
### Hi @karlavdelgadof , 

**Great work on making the changes requested by a previous reviewer 👏🏻** 

Let me point out:
- Project is properly setup
- Tests are running successfully 
- Using redux

Your project is complete! . There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

5SW

**Cheers and Happy coding! :clap: :clap: :clap:**

Feel free to leave any questions or comments in the PR thread by mentioning my Github username if something is not 100% clear.
***
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have  limited number of  reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**